packages/sdk/src/chains/evm.ts

```diff
@@ -515,102 +522,178 @@ export class EvmChain implements IChain {
+					stateDiffs.push({ slot: balanceSlot as HexString, value: testValue })
 
-						return { address: tokenAddress, stateDiff: stateDiffs }
+					try {
+						const allowanceSlot = await getStorageSlot(
+							destChain.publicClient,
+							tokenAddress,
+							1,
+							MOCK_ADDRESS,
+							intentGatewayAddress,
+						)
+						stateDiffs.push({ slot: allowanceSlot as HexString, value: testValue })
 					} catch (e) {
-						console.warn(`Could not find balance slot for token ${tokenAddress}`, e)
-						return null
+						console.warn(`Could not find allowance slot for token ${tokenAddress}:`, e)
 					}
-				}),
-			)
-		).filter(Boolean)
 
-		const fillGas = await this.publicClient.estimateContractGas({
+					return { address: tokenAddress, stateDiff: stateDiffs }
+				} catch (e) {
+					console.warn(`Could not find balance slot for token ${tokenAddress}:`, e)
+					return null
+				}
+			}),
+		).then((results) => results.filter(Boolean))
+
+		const { address: feeTokenAddress } = await destChain.getFeeTokenWithDecimals()
+		const feeTokenBalanceSlot = await getStorageSlot(destChain.publicClient, feeTokenAddress, 0, MOCK_ADDRESS)
+		const feeTokenAllowanceSlot = await getStorageSlot(
+			destChain.publicClient,
+			feeTokenAddress,
+			1,
+			MOCK_ADDRESS,
+			intentGatewayAddress,
+		)
+		const feeTokenStateDiffs = [
+			{ slot: feeTokenBalanceSlot, value: testValue },
+			{ slot: feeTokenAllowanceSlot, value: testValue },
+		]
+
+		orderOverrides.push({
+			address: feeTokenAddress,
+			stateDiff: feeTokenStateDiffs as any,
+		})
+
+		const destChainFillGas = await destChain.publicClient.estimateContractGas({
 			abi: IntentGateway.ABI,
 			address: intentGatewayAddress,
 			functionName: "fillOrder",
 			args: [transformOrderForContract(order), fillOptions as any],
-			account: fillerWalletAddress,
+			account: MOCK_ADDRESS,
 			value: totalEthValue,
-			stateOverride: overrides as any,
+			stateOverride: orderOverrides as any,
 		})
 
-		const swapOperations = await this.calculateSwapOperations({
-			order,
-			fillerWalletAddress,
-			availableAssets,
-			universalRouterAddress,
-			venues,
-		})
+		let fillGasFeeToken = await this.convertGasToFeeToken(destChainFillGas, destChain.publicClient)
+		let protocolFee = await this.getProtocolFee(order, relayerFeeInFeeToken) // Already in fee token but in destination chain decimals
 
-		const relayerFeeGas = postGasEstimate + (postGasEstimate * BigInt(200)) / BigInt(10000)
+		// Adjustments
+		fillGasFeeToken = this.adjustRelayerFeeDecimals(
+			fillGasFeeToken,
+			destChainFeeTokenDecimals,
+			sourceChainFeeTokenDecimals,
+		)
+		protocolFee = this.adjustRelayerFeeDecimals(protocolFee, destChainFeeTokenDecimals, sourceChainFeeTokenDecimals)
+		relayerFeeInFeeToken = this.adjustRelayerFeeDecimals(
+			relayerFeeInFeeToken,
+			destChainFeeTokenDecimals,
+			sourceChainFeeTokenDecimals,
+		)
```

Comment:
I feel like this could be made easier to read. 

Just get the fill gas estimate in source chain feetoken decimals, no need for converting it about